frontend/test/metabase/scenarios/question/notebook.cy.spec.js

```diff
@@ -133,6 +133,55 @@ describe("scenarios > question > notebook", () => {
     });
   });
 
+  it("should append indexes to duplicate custom expression names", () => {
+    cy.intercept("POST", "/api/dataset").as("dataset");
+    openProductsTable({ mode: "notebook" });
+
+    cy.findByText("Custom column").click();
+    popover()
+      .findByText("Category")
+      .click();
+    cy.findByPlaceholderText("Something nice and descriptive")
+      .click()
+      .type("EXPR");
+    cy.button("Done").click();
+
+    cy.findByTestId("step-expression-0-0").within(() => {
+      cy.icon("add").click();
+    });
+    popover()
+      .findByText("Category")
+      .click();
+    cy.findByPlaceholderText("Something nice and descriptive")
+      .click()
+      .type("EXPR");
+    cy.button("Done").click();
```

Comment:
Nit: This sequence is repeating enough times. What do you think about extracting it into a helper function to make this test more succint?